sdk/basyx/aas/model/base.py

```
@@ -454,25 +454,35 @@ def from_referable(referable: "Referable") -> "Key":
         """
         # Get the `type` by finding the first class from the base classes list (via inspect.getmro), that is contained
         # in KEY_ELEMENTS_CLASSES
-        from . import KEY_TYPES_CLASSES, SubmodelElementList
+        from . import SubmodelElementList
+        key_type = Key._get_key_type_for_referable(referable)
+        key_value = Key._get_key_value_for_referable(referable)
+        return Key(key_type, key_value)
+
+    @staticmethod
+    def _get_key_type_for_referable(referable: "Referable") -> KeyTypes:
+        from . import KEY_TYPES_CLASSES, resolve_referable_class_in_key_types
         try:
-            key_type = next(iter(KEY_TYPES_CLASSES[t]
-                                 for t in inspect.getmro(type(referable))
-                                 if t in KEY_TYPES_CLASSES))
+            ref_type = resolve_referable_class_in_key_types(referable)
+            key_type = KEY_TYPES_CLASSES[ref_type]
         except StopIteration:
             key_type = KeyTypes.PROPERTY
```

Does that mean, if we cannot determine the `key_type`, we cast it to be a `KeyTypes.PROPERTY`? 

That seems weird, why would we do this? Shouldn't we rather raise an exception here? Something like: `raise TypeError(f"Could not determine KeyType for Referable {referable} with type {ref_type}, unknown KeyType.", e)`